packages/thirdweb/src/wallets/wallet-connect/index.ts

```diff
@@ -521,3 +499,45 @@ async function getRequestedChainsIds(): Promise<number[]> {
   const data = localStorage.getItem(storageKeys.requestedChains);
   return data ? JSON.parse(data) : [];
 }
+
+type ArrayOneOrMore<T> = {
+  0: T;
+} & Array<T>;
+
+function getChainsToRequest(options: {
+  chain?: Chain;
+  optionalChains?: Chain[];
+  client: ThirdwebClient;
+}) {
+  const rpcMap: Record<number, string> = {};
+
+  if (options.chain) {
+    rpcMap[options.chain.id] = getRpcUrlForChain({
+      chain: options.chain,
+      client: options.client,
+    });
+  }
+
+  // limit optional chains to 10
+  const optionalChains = (options?.optionalChains || []).slice(0, 10);
```

Comment:
was this a limitation from WC? or our choice?

Reply:
Wallet connection does not work if send too many optional chains ( this happens in our dashboard - we also limit the optional chains in v4 )